app/src/main/java/de/westnordost/streetcomplete/quests/surface/DetailRoadSurfaceForm.kt

```diff
@@ -0,0 +1,105 @@
+package de.westnordost.streetcomplete.quests.surface
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.EditText
+import androidx.appcompat.app.AlertDialog
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.quests.AGroupedImageListQuestAnswerFragment
+import de.westnordost.streetcomplete.quests.OtherAnswer
+import de.westnordost.streetcomplete.util.TextChangedWatcher
+import de.westnordost.streetcomplete.view.Item
+import de.westnordost.streetcomplete.quests.surface.Surface.*
+
+class DetailRoadSurfaceForm  : AGroupedImageListQuestAnswerFragment<String, DetailSurfaceAnswer>() {
+
+    override val topItems get() =
+        if (osmElement!!.tags["surface"] == "paved")
+            listOf(ASPHALT, CONCRETE, SETT, PAVING_STONES, WOOD, GRASS_PAVER).toItems()
+        else
+            listOf(DIRT, GRASS, PEBBLES, FINE_GRAVEL, SAND, COMPACTED).toItems()
+
+    // note that for unspecific groups null is used as a value, it makes them unselecteable
+    override val allItems = listOf(
+            Item(null, R.drawable.panorama_surface_paved, R.string.quest_surface_value_paved, null, pavedSurfaces().toItems()),
+            Item(null, R.drawable.panorama_surface_unpaved, R.string.quest_surface_value_unpaved, null, unpavedSurfaces().toItems()),
+            Item(null, R.drawable.panorama_surface_ground, R.string.quest_surface_value_ground, null, groundSurfaces().toItems())
+    )
```

Comment:
Wait, why are in the `allItems` all the categories, even though we are not interested at all in the categories?

Reply:
You should probably check out the class `AImageListQuestAnswerFragment`. We do not actually want any grouping in the `DetailRoadSurfaceForm`

Reply:
> Wait, why are in the allItems all the categories, even though we are not interested at all in the categories?

to keep interface as close as possible to other surface quests.

> We do not actually want any grouping in the DetailRoadSurfaceForm

Grouping will disappear soon.

Reply:
What do you mean, it will disappear soon? Anyway, it would be better if you used `AImageListQuestAnswerFragment`